orchagent/routeorch.cpp

```diff
@@ -1473,6 +1501,18 @@ bool RouteOrch::addRoute(RouteBulkContext& ctx, const NextHopGroupKey &nextHops)
                 /* Return false since the original route is not successfully added */
                 return false;
             }
+            else
```

Comment:
can you have 'else if m_neighborToResolve is  not empty' instead of 'else'. Basically want to avoid this loop for every route addition if there is no such unresolved neighbors

Reply:
Moved this part to addNextHop so that this logic is called only once for each neighbor.